src/main/java/org/elasticsearch/index/query/TermsFilterParser.java

```diff
@@ -188,7 +188,12 @@ public Filter parse(QueryParseContext parseContext) throws IOException, QueryPar
 
             // external lookup, use it
             TermsLookup termsLookup = new TermsLookup(lookupIndex, lookupType, lookupId, lookupRouting, lookupPath, parseContext, SearchContext.current());
-            terms.addAll(termsFilterCache.terms(termsLookup, lookupCache, cacheKey));
```

Comment:
This is an unnecessary copy that I replace with just a variable assignment.  This is a slowdown in step (1), fetching the terms.